benchmarking/repos/hg.py

```diff
@@ -56,16 +56,15 @@ def getCommitTime(self, commit):
         return int(float(t[start:end]))
 
     def getNextCommitHash(self, commit, step):
-        commit_str = self._run('log', '-T', '<START>{date|hgdate}<END>', '-r', commit)
-        start = commit_str.index('<START>') + len('<START>')
-        end = commit_str.index('<END>')
-        commit_date = commit_str[start:end].split()
-        next_commit_date = int(commit_date[0]) + step
-
-        commit_str = self._run('log', '-d', '<' + str(next_commit_date) + ' ' + commit_date[1], '-l', '1', '--template', '<START>{node}<END>')
-        start = commit_str.index('<START>') + len('<START>')
-        end = commit_str.index('<END>')
-        return commit_str[start:end]
+        self.pull(commit)
+        res = self._run('next', str(step))
```

Comment:
so is this function just based on time? and it will not go to branch?

Reply:
no, it will be based on the next X commits. The assumption is that if the user do this in the master branch (no local change), everything should be serialized.